modular_skyrat/modules/customization/modules/mob/living/carbon/human/species/hemophage.dm

```diff
@@ -23,68 +23,242 @@
+	var/bloodloss_speed_multiplier = 1
 	veteran_only = TRUE
 
+
 /datum/species/hemophage/check_roundstart_eligible()
-	if(SSevents.holidays && SSevents.holidays[HALLOWEEN]) // SKYRAT EDIT - sleepy time roundstart check
+	if(SSevents.holidays && SSevents.holidays[HALLOWEEN])
 		return TRUE
+
 	return ..()
 
-/datum/species/hemophage/on_species_gain(mob/living/carbon/human/new_vampire, datum/species/old_species)
+
+/datum/species/hemophage/on_species_gain(mob/living/carbon/human/new_hemophage, datum/species/old_species)
 	. = ..()
-	to_chat(new_vampire, "[info_text]")
-	new_vampire.update_body(0)
-	new_vampire.set_safe_hunger_level()
+	to_chat(new_hemophage, "[info_text]")
+	new_hemophage.update_body(0)
+	new_hemophage.set_safe_hunger_level()
 	if(SSevents.holidays && SSevents.holidays[HALLOWEEN])
 		halloween_version = TRUE
 
-/datum/species/hemophage/spec_life(mob/living/carbon/human/vampire, delta_time, times_fired)
+
+/datum/species/hemophage/spec_life(mob/living/carbon/human/hemophage, delta_time, times_fired)
 	. = ..()
-	if(vampire.stat == DEAD)
+	if(hemophage.stat == DEAD)
 		return
-	if(istype(vampire.loc, /obj/structure/closet) && !istype(vampire.loc, /obj/structure/closet/body_bag))
-		vampire.heal_overall_damage(1.5 * delta_time, 1.5 * delta_time, 0, BODYTYPE_ORGANIC) // Fast, but not as fast due to them being able to use normal lockers.
-		vampire.adjustToxLoss(-1 * delta_time) // 50% base speed to keep it fair.
-		vampire.adjustOxyLoss(-2 * delta_time)
-		vampire.adjustCloneLoss(-0.5 * delta_time) // HARDMODE DAMAGE
+
+	if(istype(hemophage.loc, /obj/structure/closet) && !istype(hemophage.loc, /obj/structure/closet/body_bag))
+		hemophage.heal_overall_damage(1.5 * delta_time, 1.5 * delta_time, 0, BODYTYPE_ORGANIC) // Fast, but not as fast due to them being able to use normal lockers.
+		hemophage.adjustToxLoss(-1 * delta_time) // 50% base speed to keep it fair.
+		hemophage.adjustOxyLoss(-2 * delta_time)
+		hemophage.adjustCloneLoss(-0.5 * delta_time) // HARDMODE DAMAGE
 		return
-	vampire.blood_volume -= 0.125 * delta_time
-	if(vampire.blood_volume <= BLOOD_VOLUME_SURVIVE)
-		to_chat(vampire, span_danger("You ran out of blood!"))
-		vampire.death() // Owch! Ran out of blood.
-	var/area/A = get_area(vampire)
-	if(istype(A, /area/station/service/chapel) && halloween_version) // If hemophages have bat form, they cannot enter the church
-		to_chat(vampire, span_warning("You don't belong here!"))
-		vampire.adjustFireLoss(10 * delta_time)
-		vampire.adjust_fire_stacks(3 * delta_time)
-		vampire.ignite_mob()
+
+	hemophage.blood_volume -= 0.125 * delta_time * bloodloss_speed_multiplier
+
+	if(hemophage.blood_volume <= BLOOD_VOLUME_SURVIVE)
+		to_chat(hemophage, span_danger("You ran out of blood!"))
+		hemophage.death() // Owch! Ran out of blood.
+
+	if(halloween_version)// If hemophages have bat form, they cannot enter the church
+		var/area/current_area = get_area(hemophage)
+		if(istype(current_area, /area/station/service/chapel))
```

Comment:
you can just make this `if(istype(get_area(hemophage), /area/station/service/chapel))`